packages/flame/lib/src/components/component.dart

```diff
@@ -131,34 +131,28 @@ class Component with Loadable {
     children.forEach((c) => c.update(dt));
   }
 
-  @mustCallSuper
-  void render(Canvas canvas) {
-    preRender(canvas);
-  }
+  void render(Canvas canvas) {}
 
-  @mustCallSuper
   void renderTree(Canvas canvas) {
+    preRender(canvas);
     render(canvas);
-    postRender(canvas);
-    children.forEach((c) {
-      canvas.save();
-      c.renderTree(canvas);
-      canvas.restore();
-    });
+    children.forEach((c) => c.renderTree(canvas));
 
     // Any debug rendering should be rendered on top of everything
     if (debugMode) {
       renderDebugMode(canvas);
```

Comment:
If the canvas is flipped, won't the text in the debug mode also be flipped then?
Maybe that was the case before too.

Reply:
It was the same before, only then `canvas.save()/restore()` were surrounding the `renderTree()` call, now they are inside the `renderTree()` call, but still surround the rest of the content there.